src/openrct2/interface/Screenshot.cpp

```diff
@@ -337,23 +276,45 @@ static void ReleaseDPI(rct_drawpixelinfo& dpi)
     dpi.height = 0;
 }
 
-static rct_viewport GetGiantViewport(const TileCoordsXY& mapSize, int32_t rotation, ZoomLevel zoom)
+static rct_viewport GetGiantViewport(int32_t rotation, ZoomLevel zoom)
 {
     // Get the tile coordinates of each corner
-    auto leftTileCoords = GetEdgeTile(mapSize, rotation, EdgeType::LEFT, false);
-    auto rightTileCoords = GetEdgeTile(mapSize, rotation, EdgeType::RIGHT, false);
-    auto bottomTileCoords = GetEdgeTile(mapSize, rotation, EdgeType::BOTTOM, false);
+    const TileCoordsXY cornerCoords[2][4] = {
+        {
+            // Map corners
+            { 1, 1 },
+            { gMapSize.x - 2, gMapSize.y - 2 },
+            { 1, gMapSize.y - 2 },
+            { gMapSize.x - 2, 1 },
+        },
+        {
+            // Horizontal view clipping corners
+            TileCoordsXY{ gClipSelectionA },
+            TileCoordsXY{ gClipSelectionB },
+            TileCoordsXY{ CoordsXY{ gClipSelectionA.x, gClipSelectionB.y } },
+            TileCoordsXY{ CoordsXY{ gClipSelectionB.x, gClipSelectionA.y } },
+        },
+    };
 
-    // Centre the coordinates so we don't have a hard crop at the edge of the visible tile
-    leftTileCoords += CoordsXY(16, 16);
-    rightTileCoords += CoordsXY(16, 16);
-    bottomTileCoords += CoordsXY(16, 16);
+    auto* const mainWindow = window_get_main();
+    const auto* const mainViewport = window_get_viewport(mainWindow);
+    bool useViewClipping = (mainViewport != nullptr && mainViewport->flags & VIEWPORT_FLAG_CLIP_VIEW);
 
+    if (gClipSelectionA.x == 0 && gClipSelectionA.y == 0 && gClipSelectionB.x >= 32000 && gClipSelectionB.y >= 32000)
+    {
+        useViewClipping = false;
+    }
```

Comment:
This is not right, it disables vertical clipping too.

Reply:
I am using a separate bool now to check if that selection is way too large.
Then perform the check at auto corners. 
If both bools are true, then the limiteed corner selection is used. Otherwise use the entire map.

This change leaves useViewClipping untouched, meaning that vertical clipping will remain enabled.